packages/server/src/automations/triggers.ts

```diff
@@ -171,3 +203,28 @@ export async function rebootTrigger() {
     })
   }
 }
+
+async function checkTriggerFilters(
+  automation: Automation,
+  event: { row: Row; oldRow: Row }
+): Promise<boolean> {
+  const trigger = automation.definition.trigger
+  const filters = trigger?.inputs?.filters
+  const tableId = trigger?.inputs?.tableId
+
+  if (!filters) {
+    return true // No filters or tableId, so trigger by default
+  }
+
+  const newRow = await automationUtils.cleanUpRow(tableId, event.row)
+  const newRowPasses = rowPassesFilters(newRow, filters)
+
+  if (
+    trigger.stepId === definitions.ROW_UPDATED.stepId ||
+    trigger.stepId === definitions.ROW_SAVED.stepId
+  ) {
+    return newRowPasses
+  }
+
+  return true
```

Comment:
```suggestion
  if (
    trigger.stepId === definitions.ROW_UPDATED.stepId ||
    trigger.stepId === definitions.ROW_SAVED.stepId
  ) {
      const newRow = await automationUtils.cleanUpRow(tableId, event.row)
      return rowPassesFilters(newRow, filters)
  }

  return true
```

I'd restructure this like so to avoid doing work that isn't necessary.